tensorpack/dataflow/imgaug/misc.py

```diff
@@ -196,3 +196,53 @@ def _augment_coords(self, coords, do):
         if do:
             coords = coords[:, ::-1]
         return coords
+
+class RandomCropWithPadding(ImageAugmentor):
```

Comment:
This seems to be equivalent to `CenterPaste(background_filler=constant) + RandomCrop()`. But I may be mistaken.

Reply:
because the image size in pascalVOC2012 are all different, some are large, some are very small,  RandomCropWithPadding differs from CenterPaste(background_filler=constant) + RandomCrop() in :

if the image size larger than crop_size,  it will not do padding, then cropping.
else  it will do padding, then cropping.

anyway, I just move RandomCropWithPadding into the deeplabv2 example in case of pollution. if we finally find an alternative solution based on current tensorpack kernel, or this method really necessary for some dataset, we can move it into tensorpack kernel again.